.circleci/config.yml

```diff
@@ -676,7 +676,7 @@ jobs:
             DANGER_COMMAND: prepareBundleSizeReport
       - run:
           name: build @mui packages
-          command: pnpm lerna run --ignore @mui/icons-material --concurrency 6 --scope "@mui/*" build
+          command: pnpm nx run-many --exclude @mui/icons-material --parallel 6 --projects="@mui/*" -t build
```

Comment:
@Janpot @michaldudak Converted some of the commands to use nx instead of lerna as lerna was not resolving the dependencies correctly.

Reply:
What do you mean? Is there a bug in lerna? Does it work with concurrency set to 1? I'm a bit hesitant just casually swapping out lerna for nx in this PR.

Reply:
I've noticed an issue with Lerna not taking project dependencies into account as well. I'm going to investigate it further.

Reply:
@Janpot If you want, I can create a standalone PR to swap lerna with nx before merging this PR.
The main issue here is lerna not building package1 first before trying to build package2 that has a dependency on package1

Reply:
One thing I noticed while running the command `lerna ls --graph` is that it does not list out any zero runtime packages which seems weird to me even though it lists `@mui/material-nextjs` even though it was added recently. Not sure what it looks for when running this command. Here's the output - 

<details>
  <summary>Output</summary>

```json
{
  "@mui/base": [
    "@babel/runtime",
    "@floating-ui/react-dom",
    "@mui/types",
    "@mui/utils",
    "@popperjs/core",
    "clsx",
    "prop-types"
  ],
  "@mui/codemod": [
    "@babel/core",
    "@babel/runtime",
    "@babel/traverse",
    "jscodeshift",
    "jscodeshift-add-imports",
    "yargs"
  ],
  "@mui/core-downloads-tracker": [],
  "@mui/docs": [
    "@babel/runtime",
    "@mui/base",
    "@mui/utils",
    "nprogress",
    "prop-types"
  ],
  "@mui/envinfo": [
    "envinfo"
  ],
  "@mui/icons-material": [
    "@babel/runtime"
  ],
  "@mui/joy": [
    "@babel/runtime",
    "@mui/base",
    "@mui/core-downloads-tracker",
    "@mui/system",
    "@mui/types",
    "@mui/utils",
    "clsx",
    "prop-types"
  ],
  "@mui/lab": [
    "@babel/runtime",
    "@mui/base",
    "@mui/system",
    "@mui/types",
    "@mui/utils",
    "clsx",
    "prop-types"
  ],
  "@mui/material": [
    "@babel/runtime",
    "@mui/base",
    "@mui/core-downloads-tracker",
    "@mui/system",
    "@mui/types",
    "@mui/utils",
    "@types/react-transition-group",
    "clsx",
    "csstype",
    "prop-types",
    "react-is",
    "react-transition-group"
  ],
  "@mui/material-next": [
    "@babel/runtime",
    "@mui/base",
    "@mui/material",
    "@mui/system",
    "@mui/types",
    "@mui/utils",
    "@types/react-transition-group",
    "clsx",
    "prop-types",
    "react-is",
    "react-transition-group"
  ],
  "@mui/material-nextjs": [],
  "@mui/private-theming": [
    "@babel/runtime",
    "@mui/utils",
    "prop-types"
  ],
  "@mui/styled-engine": [
    "@babel/runtime",
    "@emotion/cache",
    "csstype",
    "prop-types"
  ],
  "@mui/styled-engine-sc": [
    "@babel/runtime",
    "csstype",
    "hoist-non-react-statics",
    "prop-types"
  ],
  "@mui/styles": [
    "@babel/runtime",
    "@emotion/hash",
    "@mui/private-theming",
    "@mui/types",
    "@mui/utils",
    "clsx",
    "csstype",
    "hoist-non-react-statics",
    "jss",
    "jss-plugin-camel-case",
    "jss-plugin-default-unit",
    "jss-plugin-global",
    "jss-plugin-nested",
    "jss-plugin-props-sort",
    "jss-plugin-rule-value-function",
    "jss-plugin-vendor-prefixer",
    "prop-types"
  ],
  "@mui/system": [
    "@babel/runtime",
    "@mui/private-theming",
    "@mui/styled-engine",
    "@mui/types",
    "@mui/utils",
    "clsx",
    "csstype",
    "prop-types"
  ],
  "@mui/types": [],
  "@mui/utils": [
    "@babel/runtime",
    "@types/prop-types",
    "prop-types",
    "react-is"
  ]
}
```
lerna success found 18 packages
</details>

Reply:
Removed this now.